usaspending_api/disaster/v2/views/award/amount.py

```diff
@@ -40,13 +47,41 @@ def post(self, request):
         if all(x in self.filters for x in ["award_type_codes", "award_type"]):
             raise UnprocessableEntityException("Cannot provide both 'award_type_codes' and 'award_type'")
 
-        self.nonzero_fields = ["obligated_sum", "outlay_sum"]
+        award_type = self.filters.get("award_type")
 
-        if self.award_type_codes and set(self.award_type_codes) <= set(loan_type_mapping.keys()):
-            self.nonzero_fields.append("total_loan_value")
+        queryset = (
+            CovidFABASpending.objects.filter(spending_level="awards")
+            .filter(defc__in=self.filters["def_codes"])
+            .annotate(
+                total_award_count=Sum("award_count"),
+                total_obligation_sum=Sum("obligation_sum"),
+                total_outlay_sum=Sum("outlay_sum"),
+                total_face_value_of_loan=Sum("face_value_of_loan"),
+            )
+        )
+
+        if self.award_type_codes:
+            queryset = queryset.filter(award_type__in=self.award_type_codes)
+
+        if award_type and award_type.lower() == "procurement":
```

Comment:
award_type is not a boolean. So coding `if award_type` to check if the variable equals `None` is implicit. This predicate can be improved to be more explicit. Try `if award_type is not None`

Reply:
Updated the `None` checks to be explicit.